File: 3172-divisible-and-non-divisible-sums-difference/3172-divisible-and-non-divisible-sums-difference.ts
function differenceOfSums(n: number, m: number): number {
    const total_sum = (n * (n + 1)) / 2;
    const pairs = Math.floor(n / m);
    const divisble_sum = m * (pairs * (pairs + 1)) / 2;
    const non_divisble_sum = total_sum - divisble_sum;
    return non_divisble_sum - divisble_sum;
};

File: 3172-divisible-and-non-divisible-sums-difference/3172-divisible-and-non-divisible-sums-difference.java
class Solution {
    public int differenceOfSums(int n, int m) {
        int totalSum = (n * (n + 1)) / 2;
        int pairs = (int) Math.floor(n / m);
        int divisbleSum = m * (pairs * (pairs + 1)) / 2;
        return totalSum - 2 * divisbleSum;
    }
}